### PR TITLE
Add support for root folder icons.

### LINKED ISSUE #1168: [Task] Support root folder icons
With the introduction of multi-workspace `root-folder` in `vscode`, we should support specifying an icon for that.

### PR DESCRIPTION
**Pull request by JimiC**
***Fixes #1168***

**Changes proposed:**
* [x] Add

**Things I've done:**
* [x] My pull request fixes an issue, I referenced the issue.

I have also done some refactoring in the `iconSchema` models. Any objections are welcome.

Please test this out and point me to any misbehavior or lack of testing cases.

~This PR will also requires additional translations before it gets merged.~

### CODE REVIEW

File: package.nls.de.json [hunk 1]
```diff
@@ -26,5 +26,7 @@
   "configuration.associations.fileDefault.file.description": "Diese Einstellung macht es dir möglich, ein benutzerdefiniertes Datei-Icon für dunkle Themes zu verwenden.",
   "configuration.associations.fileDefault.file_light.description": "Diese Einstellung macht es dir möglich, ein benutzerdefiniertes Datei-Icon für helle Themes zu verwenden.",
   "configuration.associations.folderDefault.folder.description": "Diese Einstellung macht es dir möglich, ein benutzerdefiniertes Ordner-Icon für dunkle Themes zu verwenden.",
-  "configuration.associations.folderDefault.folder_light.description": "Diese Einstellung macht es dir möglich, ein benutzerdefiniertes Ordner-Icon für helle Themes zu verwenden."
+  "configuration.associations.folderDefault.root_folder.description": "",
+  "configuration.associations.folderDefault.folder_light.description": "Diese Einstellung macht es dir möglich, ein benutzerdefiniertes Ordner-Icon für helle Themes zu verwenden.",
+  "configuration.associations.folderDefault.root_folder_light.description": ""
```
Comment (jens1o) [inline package.nls.de.json]: `Diese Einstellung macht es dir möglich, ein benutzerdefiniertes helles Ordner-Icon für Wurzelverzeichnisse festzulegen.`

File: package.nls.de.json [hunk 2]
```diff
@@ -26,5 +26,7 @@
   "configuration.associations.fileDefault.file.description": "Diese Einstellung macht es dir möglich, ein benutzerdefiniertes Datei-Icon für dunkle Themes zu verwenden.",
   "configuration.associations.fileDefault.file_light.description": "Diese Einstellung macht es dir möglich, ein benutzerdefiniertes Datei-Icon für helle Themes zu verwenden.",
   "configuration.associations.folderDefault.folder.description": "Diese Einstellung macht es dir möglich, ein benutzerdefiniertes Ordner-Icon für dunkle Themes zu verwenden.",
-  "configuration.associations.folderDefault.folder_light.description": "Diese Einstellung macht es dir möglich, ein benutzerdefiniertes Ordner-Icon für helle Themes zu verwenden."
+  "configuration.associations.folderDefault.root_folder.description": "",
```
Comment (jens1o) [inline package.nls.de.json]: `Diese Einstellung macht es dir möglich, ein benutzerdefiniertes dunkles Ordner-Icon für Wurzelverzeichnisse festzulegen.`

File: src/models/iconSchema/iconDefinition.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  _folder: IIconPath;
+  _folder_open: IIconPath;
+  _file_light: IIconPath;
+  _folder_light: IIconPath;
+  _folder_light_open: IIconPath;
+  _root_folder: IIconPath;
+  _root_folder_open: IIconPath;
+  _root_folder_light: IIconPath;
+  _root_folder_light_open: IIconPath;
+
```
Comment (jens1o) [inline src/models/iconSchema/iconDefinition.ts]: is that whitespace by design?

Comment (JimiC) [reply src/models/iconSchema/iconDefinition.ts]: No. Will fix it tomorrow along with the test failure.

Comment (JimiC) [reply src/models/iconSchema/iconDefinition.ts]: Fixed.

**Comment by robertohuertasm** [comment]
I personally don't like this green color of the `slash`. I would prefer something not as visually invasive. Something similar to this:

image

Just a suggestion.

**Comment by JimiC** [comment]
Can you provide the color hash?

**Comment by robertohuertasm** [comment]
#FFEEBE

**Comment by JimiC** [comment]
I have the feeling that this color will not show good on light themes.

**Comment by robertohuertasm** [comment]
It's not a feeling but a certainty! I forgot to test it with light themes. I've tried adding a 0.5pt border to the `slash` and I think it works. It's very subtle in dark themes but noticeable in light themes:

image

image

**Comment by JimiC** [comment]
Or we could use the glow effect. Will try that tomorrow.

File: package.nls.es.json [hunk 4]
```diff
@@ -26,5 +26,7 @@
   "configuration.associations.fileDefault.file.description": "Este ajuste permite cambiar el icono de archivo por defecto para los temas oscuros.",
   "configuration.associations.fileDefault.file_light.description": "Este ajuste permite cambiar el icono de archivo por defecto para los temas claros.",
   "configuration.associations.folderDefault.folder.description": "Este ajuste permite cambiar el icono de carpeta por defecto para los temas oscuros.",
-  "configuration.associations.folderDefault.folder_light.description": "Este ajuste permite cambiar el icono de carpeta por defecto para los temas claros."
+  "configuration.associations.folderDefault.root_folder.description": "",
```
Comment (robertohuertasm) [inline package.nls.es.json]: Este ajuste permite cambiar el icono de carpeta raíz por defecto para los temas oscuros.

File: package.nls.es.json [hunk 5]
```diff
@@ -26,5 +26,7 @@
...
+  "configuration.associations.folderDefault.folder_light.description": "Este ajuste permite cambiar el icono de carpeta por defecto para los temas claros.",
```
Comment (robertohuertasm) [inline package.nls.es.json]: Este ajuste permite cambiar el icono de carpeta raíz por defecto para los temas claros.

**Comment by JimiC** [comment]
@robertohuertasm I ended using your proposed art.

**Comment by JimiC** [comment]
Please be so kind and provide your translations.

- [ ] French: @MonsieurMan
- [ ] Italian: @mariotsi @sabinrimni
- [ ] Russian: @ksvitkovsky
- [ ] Simplified Chinese: @iFun @g-plane

**Comment by iFun** [comment]
"configuration.associations.folderDefault.root_folder.description": "这项设置会更改默认的根目录文件夹图标",
 "configuration.associations.folderDefault.folder_light.description": "这项设置会更改浅色主题默认的文件夹图标",
 "configuration.associations.folderDefault.root_folder_light.description": "这项设置会更改浅色主题默认的根目录文件夹图标"

**Comment by sabinrimni** [comment]
Italian Translation:
`"configuration.associations.folderDefault.root_folder.description": "Questa impostazione ti permette di cambiare l'icona predefinita per la cartella root nei temi scuri.",
"configuration.associations.folderDefault.root_folder_light.description": "Questa impostazione ti permette di cambiare l'icona predefinita per la cartella root nei temi chiari."`

File: package.nls.fr.json [hunk 6]
```diff
@@ -26,5 +26,7 @@
   "configuration.associations.fileDefault.file.description": "Définir l'icône de fichier par défaut pour le thème sombre.",
   "configuration.associations.fileDefault.file_light.description": "Définir l'icône de fichier par défaut pour le thème clair.",
   "configuration.associations.folderDefault.folder.description": "Définir l'icône de dossier par défaut pour le thème sombre.",
-  "configuration.associations.folderDefault.folder_light.description": "Définir l'icône de dossier par défaut pour le thème clair."
+  "configuration.associations.folderDefault.root_folder.description": "",
```
Comment (MonsieurMan) [inline package.nls.fr.json]: Définir l'icone par défaut du dossier racine pour le thème sombre.

File: package.nls.fr.json [hunk 7]
```diff
@@ -26,5 +26,7 @@
...
+  "configuration.associations.folderDefault.folder_light.description": "Définir l'icône de dossier par défaut pour le thème clair.",
+  "configuration.associations.folderDefault.root_folder_light.description": ""
```
Comment (MonsieurMan) [inline package.nls.fr.json]: Définir l'icone par défaut du dossier racine pour le thème clair.

**Comment by ksvitkovsky** [comment]
```
"configuration.associations.folderDefault.root_folder.description": "Настройка позволяет переопределить значок корневой папки по-умолчанию для темной темы",
"configuration.associations.folderDefault.root_folder_light.description": "Настройка позволяет переопределить значок корневой папки по-умолчанию для светлой темы"
```

**Comment by JimiC** [comment]
This PR is ready for final review.

**Pull request merged**